src/mariadb_connection.c

```diff
@@ -248,14 +248,14 @@ MrdbConnection_Initialize(MrdbConnection *self,
   /* Todo: we need to support all dsn parameters, the current
            implementation is just a small subset.
   */
-  char *dsn= NULL, *host=NULL, *user= NULL, *password= NULL, *schema= NULL,
+  char *dsn= NULL, *host= "127.0.0.1", *user= NULL, *password= NULL, *schema= NULL,
```

Comment:
NULL value for host is handled inside MySQL Connector/C. Please note that host 127.0.0.1 will use a tcp connection, while localhost or NULL will use socket (=default).